lib/modules/datasource/julia-pkg-server/index.ts

```diff
@@ -0,0 +1,112 @@
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { Result } from '../../../util/result';
+import { joinUrlParts } from '../../../util/url';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import {
+  PKG_SERVER_REQUEST_HEADERS,
+  juliaPkgServerDatasourceId,
+} from './common';
+import {
+  cacheKeyFromRegistryUrl,
+  extractFilesFromTarball,
+  parseRegistryUrl,
+  registryPathForPackage,
+} from './registry';
+import { Package, Versions } from './schema';
+
+/** The files in a registry defining the metadata for a package. */
+type PackageMetadataPath = `${string}/Package.toml` | `${string}/Versions.toml`;
+
+export class JuliaPkgServerDatasource extends Datasource {
+  // This URL will redirect to a suitable, geographically near, mirror
+  static readonly generalRegistryUrl =
+    'https://pkg.julialang.org/registry/23338594-aafe-5451-b93e-139f81909106';
+
+  static readonly id = juliaPkgServerDatasourceId;
+
+  constructor() {
+    super(JuliaPkgServerDatasource.id);
+  }
+
+  override readonly defaultRegistryUrls = [
+    JuliaPkgServerDatasource.generalRegistryUrl,
+  ];
+
+  override readonly customRegistrySupport = true;
+
+  // Multiple package servers can contain the same package, but contain
+  // different versions of that package
+  override readonly registryStrategy = 'merge';
+
+  @cache({
+    namespace: `datasource-${juliaPkgServerDatasourceId}-releases`,
+    key: ({ packageName }: GetReleasesConfig) => packageName,
```

Comment:
Shouldn't caching be unique per registry url?

Reply:
Resolved in 77d0ec6e99c38afd9f5cb551bb6471c88496c46b.